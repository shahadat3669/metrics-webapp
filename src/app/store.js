import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';

export const store = configureStore({
  reducer: {},
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
  devTools: process.env.NODE_ENV !== 'production',
});
