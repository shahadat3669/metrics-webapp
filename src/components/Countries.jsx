import React from 'react';
import { useSelector } from 'react-redux';
import { FaArrowRight } from 'react-icons/fa';
import { Link } from 'react-router-dom';
import { getAllCountries } from '../features/countries/countriesSlice';
import styles from '../styles/components/Countries.module.scss';

const Countries = () => {
  const countries = useSelector(getAllCountries);
  return (
    <div className={styles.countries}>
      <div className={styles.countries__bar}> COUNTRIES</div>
      <div className={styles.countries__list}>
        {countries.map((country) => (
          <Link
            to={`country/${country.id}`}
            key={country.id}
            className={styles.countries__item}
          >
            <div className={styles['countries__icon-wrapper']}>
              <FaArrowRight className={styles.countries__icon} />
            </div>
            <div className={styles['countries__item-img-wrapper']}>
              <img
                className={styles['countries__item-img']}
                src={country.flag}
                alt={country.name}
              />
            </div>
            <div className={styles.countries__details}>
              <h3 className={styles['countries__item-name']}>{country.name}</h3>
              <p className={styles['countries__item-population']}>
                {country.population}
              </p>
            </div>
          </Link>
        ))}
      </div>
    </div>
  );
};

export default Countries;
