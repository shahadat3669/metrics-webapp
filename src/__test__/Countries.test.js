import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Countries from '../components/Countries';
import store from '../app/store';

describe('Rockets', () => {
  test('renders Rockets Page without crashing', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Countries />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree.toJSON).toMatchSnapshot();
  });

  test('renders Rockets Page without crashing', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Countries />
        </BrowserRouter>
      </Provider>,
    );
    container.querySelector('.mainTitle');
    expect(container.firstChild).toMatchSnapshot();

    container.querySelector('.first_color');
    expect(container.firstChild).toMatchSnapshot();
  });
});
