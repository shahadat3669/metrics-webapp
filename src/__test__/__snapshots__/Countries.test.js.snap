// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets renders Rockets Page without crashing 1`] = `undefined`;

exports[`Rockets renders Rockets Page without crashing 2`] = `
<div
  class="countries"
>
  <div
    class="countries__bar"
  >
     COUNTRIES
  </div>
  <div
    class="countries__list"
  />
</div>
`;

exports[`Rockets renders Rockets Page without crashing 3`] = `
<div
  class="countries"
>
  <div
    class="countries__bar"
  >
     COUNTRIES
  </div>
  <div
    class="countries__list"
  />
</div>
`;
